lib/country_lookup.rb

```diff
@@ -2,6 +2,9 @@ module CountryLookup
 
   class Lookup
     def initialize
+      if !@initialize_bg_thread.nil?
```

Comment:
this is a constructor in ruby, right? So wont the lookup ip string and lookup ip number functions still potentially create a new instance of this, kill in the background thread?

Reply:
Oh I see what you're saying. This needs to live outside of the class.